Core/MobileCore.cs

```diff
@@ -0,0 +1,181 @@
+﻿using System;
+using System.Collections.Generic;
+using AeroGear.Mobile.Core.Configuration;
+using AeroGear.Mobile.Core.Logging;
+using AeroGear.Mobile.Core.Exception;
+using System.Diagnostics.Contracts;
+using AeroGear.Mobile.Core.Http;
+using System.Reflection;
+using System.Net.Http;
+
+namespace AeroGear.Mobile.Core
+{
+    public sealed class MobileCore
+    {
+
+        public const String DEFAULT_CONFIG_FILE_NAME = "mobile-services.json";
+
+        private static String TAG = "AEROGEAR/CORE";
+
+        private static int DEFAULT_TIMEOUT = 20;
+
+        public ILogger Logger
+        {
+            get;
+            private set;
+        }
+
+        public String ConfigFileName
+        {
+            get;
+            private set;
+        }
+
+        private static MobileCore instance;
+
+        public static MobileCore Instance
+        {
+            get
+            {
+                if (instance == null)
+                {
+                    throw new InitializationException("Core is not initialized, don't forget to call MobileCore.init() before using this instance.");
```

Comment:
can we use `NotInitializedException` here?

Reply:
I don't know what is `NotInitializedException`. It might be some remnant.